pam/pam.go

```diff
@@ -237,15 +237,24 @@ func pam_sm_authenticate(pamh *C.pam_handle_t, flags, argc C.int, argv **C.char)
 	}
 }
 
+// cancelIsAuthorizedWatcher starts watching for a single SIGINT that will cancel the IsAuthorized call related to the
+// provided sessionID.
+//
+// This is a test function (for now) and it times out after 30 seconds.
 func cancelIsAuthorizedWatcher(client authd.PAMClient, sessionID string) {
 	ch := make(chan os.Signal, 1)
 	signal.Notify(ch, syscall.SIGINT)
 	defer signal.Stop(ch)
 
-	<-ch
-	// Ask broker to cancel current authentication mode.
-	if _, err := client.CancelIsAuthorized(context.TODO(), &authd.CIARequest{SessionId: sessionID}); err != nil {
-		log.Errorf(context.Background(), "couldn't cancel authentication mode: %v", err)
+	select {
+	case <-ch:
+		// Ask broker to cancel current authentication mode.
+		if _, err := client.CancelIsAuthorized(context.TODO(), &authd.CIARequest{SessionId: sessionID}); err != nil {
```

Comment:
If you look at the PAM module, there is a real case when you need to call CancelIsAuthorized: in the entry + wait mode, if the entry is entered and press, we need to call Cancel before reissuing IsAuthorized, mind adding it there too?